apps/web/middleware.ts

```diff
@@ -47,38 +47,45 @@ const middleware = async (req: NextRequest): Promise<NextResponse<unknown>> => {
   const requestHeaders = new Headers(req.headers);
   requestHeaders.set("x-url", req.url);
 
-  if (!url.pathname.startsWith("/api")) {
+  if (!url.pathname.startsWith("/api") && (await safeGet<boolean>("isInMaintenanceMode"))) {
     //
     // NOTE: When tRPC hits an error a 500 is returned, when this is received
     //       by the application the user is automatically redirected to /auth/login.
     //
     //     - For this reason our matchers are sufficient for an app-wide maintenance page.
     //
     // Check whether the maintenance page should be shown
-    const isInMaintenanceMode = await safeGet<boolean>("isInMaintenanceMode");
     // If is in maintenance mode, point the url pathname to the maintenance page
-    if (isInMaintenanceMode) {
-      req.nextUrl.pathname = `/maintenance`;
-      return NextResponse.rewrite(req.nextUrl);
-    }
+    req.nextUrl.pathname = `/maintenance`;
+    return NextResponse.rewrite(req.nextUrl);
   }
 
   const routingFormRewriteResponse = routingForms.handleRewrite(url);
   if (routingFormRewriteResponse) {
     return responseWithHeaders({ url, res: routingFormRewriteResponse, req });
   }
 
-  if (url.pathname.startsWith("/api/trpc/")) {
-    requestHeaders.set("x-cal-timezone", req.headers.get("x-vercel-ip-timezone") ?? "");
-  }
+  if (url.pathname.startsWith("/api/")) {
+    if (url.pathname.startsWith("/api/auth/signup") && (await safeGet<boolean>("isSignupDisabled"))) {
+      // If is in maintenance mode, point the url pathname to the maintenance page
 
-  if (url.pathname.startsWith("/api/auth/signup")) {
-    const isSignupDisabled = await safeGet<boolean>("isSignupDisabled");
-    // If is in maintenance mode, point the url pathname to the maintenance page
-    if (isSignupDisabled) {
       // TODO: Consider using responseWithHeaders here
       return NextResponse.json({ error: "Signup is disabled" }, { status: 503 });
     }
+
+    if (url.pathname.startsWith("/api/trpc/")) {
+      requestHeaders.set("x-cal-timezone", req.headers.get("x-vercel-ip-timezone") ?? "");
+    }
+
+    return responseWithHeaders({
```

Comment:
if it's an API request, we return here because we shouldn't rewrite to `/${locale}${url.pathname}`